### PR TITLE
Do not delete temp dirs

### PR DESCRIPTION
**Pull request by iam-py-test**
Advantage:
- Some apps break until a new directory (%temp% or C:\Windows\Temp) is created (i.e. dism)

Disadvantage:
- Directories in temp are not deleted

Steps to reproduce:
- Run the "Clear Windows Temp files" code
- Run `dism /online /Disable-Feature /FeatureName:"MicrosoftWindowsPowerShellV2Root" /NoRestart` (other commands may be effected)
- Note an error
- Recreate %temp% and rerun DISM
- No error

Fix:
- Replace `rd /s /q "C:\Windows\Temp"` with `del /s /q "%WINDIR%\Temp\*"`
- Delete `rd /s /q "%TEMP%"` (`del /f /q %localappdata%\Temp\*` seems to work on it's own)

Thank you. Sorry for any errors.

### CODE REVIEW

File: src/application/collections/windows.yaml [hunk 1]
```diff
@@ -581,8 +581,7 @@ actions:
                         recommend: standard
                         code: |-
                             del /f /q %localappdata%\Temp\*
-                            rd /s /q "%WINDIR%\Temp"
-                            rd /s /q "%TEMP%"
+                            del /s /q "%WINDIR%\Temp\*"
```
Comment (SnowzNZ) [inline src/application/collections/windows.yaml:584]: Possibly add /f argument to remove read only files to be consistent with line 583

**Review by SnowzNZ — CHANGES_REQUESTED** [review]
Add the `/s` argument to the line 583 command to be consistent with line 584

File: src/application/collections/windows.yaml [hunk 2]
```diff
@@ -580,8 +580,8 @@ actions:
                         name: Clear Windows temp files
                         recommend: standard
                         code: |-
-                            del /f /q %localappdata%\Temp\*
-                            del /s /q "%WINDIR%\Temp\*"
+                            del /s /f /q %TEMP%*
```
Comment (SnowzNZ) [inline src/application/collections/windows.yaml:583]: Add missing backslash to path. Should be `%TEMP%\*`

**Review by SnowzNZ — CHANGES_REQUESTED** [review]

File: src/application/collections/windows.yaml (hunk 1 above)
Comment (SnowzNZ) [reply src/application/collections/windows.yaml]: Yes all good for line 584

File: src/application/collections/windows.yaml (hunk 2 above)
Comment (iam-py-test) [reply src/application/collections/windows.yaml]: Thanks, fixed

**Review by SnowzNZ — APPROVED** [review]
Looks good to me

**Review by SnowzNZ — CHANGES_REQUESTED** [review]
Also add the prefetch folder
`del /s /f /q %WINDIR%\Prefetch\*`

**Comment by SnowzNZ** [comment]
@undergroundwires - Would be good for this to merged after the prefetch folder is added.
Confirmed [here ](https://answers.microsoft.com/en-us/windows/forum/all/temptemp-and-prefeth-files/8c3b54bb-4ca1-41c0-af22-0cd228e15498)that the prefetch folder is not important, its basically the same as the temp folder as in it is just cached files.

**Comment by iam-py-test** [comment]
Prefetch added, thanks

**Review by SnowzNZ — APPROVED** [review]

**Comment by undergroundwires** [comment]
Hi, this will be merged, thank you @iam-py-test . A quick question, why is `del /f /q %localappdata%\Temp\*` line deleted in this PR?

**Comment by iam-py-test** [comment]
Both `%localappdata%\Temp\` and `%temp%` point to the same folder. If there is some case(s) where it doesn't, I can put that line back.

**Comment by undergroundwires** [comment]
Thank you for the explanation. OK, I will merge this as it is. No need to add back localappdata. Sorry for my late response.

I also looked for some official documentation for prefetch folder and see that it's safe to delete but also good to delete in terms of privacy, good suggestion @SnowzNZ. It would be easier to handle we had different PR because prefetching is not really about "not deleting temp dirs", but no action needed, I'll this PR as it is in following days. It will fix many unintended side-effects for many, so thank you @iam-py-test for this fix.

[Recommended Test Methodology.doc | microsoft.com](https://download.microsoft.com/download/7/e/7/7e7662cf-cbea-470b-a97e-ce7ce0d98dc2/win7perf.docx):

> If you make significant system configuration changes, such as changes to drivers, services, or applications that start automatically, remove the \Windows\Prefetch directory and all its contents from the system drive. This removal eliminates any stale prefetched data. You must then retrain the system.

[Take response actions on a device in Microsoft Defender for Endpoint | Microsoft Learn](https://learn.microsoft.com/en-us/microsoft-365/security/defender-endpoint/respond-machine-alerts?view=o365-worldwide#collect-investigation-package-from-devices):

> Windows Prefetch files are designed to speed up the application startup process. It can be used to track all the files recently used in the system and find traces for applications that might have been deleted but can still be found in the prefetch file list.Prefetch folder: Contains a copy of the prefetch files from %SystemRoot%\Prefetch. NOTE: It is suggested to download a prefetch file viewer to view the prefetch files.PrefetchFilesList.txt: Contains the list of all the copied files that can be used to track if there were any copy failures to the prefetch folder.

More about prefetching: [Sysinternals Newsletter Vol. 3, No. 2 - Sysinternals | Microsoft Learn](https://learn.microsoft.com/en-us/sysinternals/resources/archive/v03n02#windows-xp-prefetching)

**Comment by undergroundwires** [comment]
Added extensive docs to these scripts https://github.com/undergroundwires/privacy.sexy/commit/d45750428cca010daf2721b33a8ae3a01b28813b. Both prefetch and the fix will be released in 0.12.5.

**Pull request merged**